client/src/showTextDocument.proposed.ts

```diff
@@ -0,0 +1,25 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import { Uri, workspace, window } from 'vscode';
+import { StaticFeature, BaseLanguageClient, ShowTextDocumentRequest } from './client';
+
+export class ShowTextDocumentFeature implements StaticFeature {
+
+	constructor(private _client: BaseLanguageClient) {
+	}
+
+	public fillClientCapabilities(): void {
```

Comment:
What is `fillClientCapabilities` for?
Is `StaticFeature` the right interface to implement for this class?

Reply:
It's for a client to declare that it supports this feature. Otherwise, you have servers sending this request to the client when the client doesn't even support this. It does not seem like you have this feature declared as a client capability in the protocol at the moment though.

Reply:
Thanks, I've declared the client capabilities!